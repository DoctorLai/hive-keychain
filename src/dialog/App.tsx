import { DialogCommand } from '@reference-data/dialog-message-key.enum';
import React, { useEffect, useState } from 'react';
import Unlock from 'src/dialog/pages/unlock';
import './dialog.scss';

const App = () => {
  useEffect(() => {
    chrome.runtime.onMessage.addListener(function (data, sender, sendResp) {
      setData(data);
    });
  }, []);

  const [data, setData] = useState<any>({});

  const renderDialogContent = () => {
    console.log(data);
    switch (data.command) {
      case DialogCommand.UNLOCK:
      case DialogCommand.WRONG_MK:
        return (
          <Unlock
            data={data}
            wrongMk={data.command === DialogCommand.WRONG_MK}
          />
        );
      default:
        return null;
    }
  };

  return <div className="dialog">{renderDialogContent()}</div>;
};

export default App;
