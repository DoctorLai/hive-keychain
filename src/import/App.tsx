import React, { Component } from 'react';
import { BackgroundCommand } from 'src/reference-data/background-message-key.enum';
import FileUtils from 'src/utils/file.utils';
import './import-keys.scss';

interface ImportAccountState {
  selectedFile?: File;
}

class App extends Component {
  state: ImportAccountState = {};

  handleFileUpload = (event: any) => {
    this.state.selectedFile = event.target.files[0];
  };

  importKeysFromFile = async () => {
    if (this.state.selectedFile) {
      const base64 = await FileUtils.toBase64(this.state.selectedFile);
      const fileData = atob(base64);
      chrome.runtime.sendMessage({
        command: BackgroundCommand.IMPORT_ACCOUNTS,
        value: fileData,
      });
    }
  };

  render() {
    return (
      <div className="import-keys">
        <div className="title-panel">
          <img src="/assets/images/iconhive.png" />
          <div className="title">
            {chrome.i18n.getMessage('import_html_title')}
          </div>
        </div>
        <div
          className="caption"
          dangerouslySetInnerHTML={{
            __html: chrome.i18n.getMessage('import_html_text'),
          }}></div>

        <input
          type="file"
          accept=".kc"
          id="file"
          onChange={this.handleFileUpload}
        />
        <label htmlFor="file">Choose a file</label>
        <span id="file_span">{this.state.selectedFile?.name}</span>
        <button id="proceed" onClick={this.importKeysFromFile}>
          {chrome.i18n.getMessage('popup_html_import')}
        </button>
      </div>
    );
  }
}

export default App;
