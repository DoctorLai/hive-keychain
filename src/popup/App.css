.App {
    height: 600px;
    width: 300px;
    font-family: Futura !important;
    font-weight: 400;
    background: url("/assets/images/hive-watermark.png") 0% 0% no-repeat
    padding-box, linear-gradient(330deg, #59626c 0%, #000000 100%) 0%
    0% no-repeat padding-box !important;

    color: white;

}