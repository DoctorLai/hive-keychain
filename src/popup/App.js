import React, { Component } from "react";

class App extends Component {
  render() {
    return (
      <div className="App">
        <h1>{chrome.i18n.getMessage("popup_html_new_password")}</h1>
      </div>
    );
  }
}

export default App;
