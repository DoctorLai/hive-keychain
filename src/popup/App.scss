@import './variable.scss';
@font-face {
  font-family: Futura;
  src: url('/assets/fonts/Futura-Boo.otf');
  font-weight: 400;
}

.App {
  height: 600px;
  width: 350px;
  font-family: Futura !important;
  font-weight: 400;
  background: url('/assets/images/hive-watermark.png') 0% 0% no-repeat
      padding-box,
    linear-gradient(330deg, #59626c 0%, #000000 100%) 0% 0% no-repeat
      padding-box !important;
  color: white;
  box-sizing: border-box;
  overflow: auto;

  &.homepage {
    background: url(/assets/images/hive-watermark.png) 0% 0% no-repeat
        padding-box,
      linear-gradient(to bottom, #000000, #59626c, #000000) 0% 0% no-repeat
        padding-box !important;
  }

  .page-container {
    display: grid;
    row-gap: 32px;
    grid-template-rows: 28px 1fr;
    height: 100%;
  }
  .divider {
    flex: 1;
  }
}

.react-dropdown-select-dropdown-handle {
  width: 48px;
  svg {
    width: 20px !important;
    height: 20px !important;
  }
}

.pretty.p-switch .state:before {
  top: calc((0% - (100% - 1em)) - 13%) !important;
}

.css-wmw4vi-ReactDropdownSelect {
  padding: 2px 0px !important;
}

.Toastify__progress-bar {
  bottom: unset;
  top: 0;
}

.Toastify__toast-icon {
  display: none !important;
}

.Toastify__progress-bar {
  background: white !important;
}

.Toastify__toast--error {
  background: $message-container-error !important;
}
.Toastify__toast--info {
  background: $message-container-info !important;
}
.Toastify__toast--success {
  background: $message-container-success !important;
}
