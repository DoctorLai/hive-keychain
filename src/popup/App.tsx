import {retrieveAccounts} from '@popup/actions/account.actions';
import {setMk} from '@popup/actions/mk.actions';
import {navigateTo} from '@popup/actions/navigation.actions';
import {RootState} from '@popup/store';
import React, {useEffect} from 'react';
import {connect, ConnectedProps} from 'react-redux';
import {BackgroundMessage} from 'src/background/background-message.interface';
import {BackgroundCommand} from 'src/reference-data/background-message-key.enum';
import {Screen} from 'src/reference-data/screen.enum';
import PopupUtils from 'src/utils/popup.utils';
import './App.css';
import {AddAccountRouterComponent} from './pages/add-account/add-account-router/add-account-router.component';
import {AppContainerComponent} from './pages/app-container/app-container.component';
import {ErrorMessageContainerComponent} from './pages/error-message-container/error-message-container.component';
import {SignInComponent} from './pages/sign-in/sign-in.component';
import {SignUpComponent} from './pages/sign-up/sign-up.component';

const App = ({
  setMk,
  mk,
  retrieveAccounts,
  accounts,
  navigateTo,
}: PropsFromRedux) => {
  useEffect(() => {
    PopupUtils.fixPopupOnMacOs();
  }, []);

  useEffect(() => {
    chrome.runtime.sendMessage({command: BackgroundCommand.GET_MK});
    chrome.runtime.onMessage.addListener(onSentBackMkListener);
  }, [setMk]);

  const onSentBackMkListener = (message: BackgroundMessage) => {
    if (message.command === BackgroundCommand.SEND_BACK_MK) {
      setMk(message.value);
      retrieveAccounts(message.value);
      chrome.runtime.onMessage.removeListener(onSentBackMkListener);
    }
  };

  const renderMainLayoutNav = () => {
    if (!mk) {
      if (accounts.length === 0) {
        return <SignUpComponent />;
      } else {
        return <SignInComponent />;
      }
    } else {
      if (accounts.length === 0) {
        navigateTo(Screen.ACCOUNT_PAGE_INIT_ACCOUNT);
        return <AddAccountRouterComponent />;
      } else {
        return <AppContainerComponent />;
      }
    }
  };

  return (
    <div className="App">
      {renderMainLayoutNav()}
      <ErrorMessageContainerComponent />
    </div>
  );
};

const mapStateToProps = (state: RootState) => {
  return {
    mk: state.mk,
    accounts: state.accounts,
  };
};

const connector = connect(mapStateToProps, {
  setMk,
  retrieveAccounts,
  navigateTo,
});
type PropsFromRedux = ConnectedProps<typeof connector>;

export default connector(App);
