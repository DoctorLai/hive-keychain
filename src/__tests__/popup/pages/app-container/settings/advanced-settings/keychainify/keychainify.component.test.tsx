import App from '@popup/App';
import { Icons } from '@popup/icons.enum';
import { LocalStorageKeyEnum } from '@reference-data/local-storage-key.enum';
import { Screen } from '@reference-data/screen.enum';
import '@testing-library/jest-dom';
import { act, cleanup, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import React from 'react';
import ariaLabelButton from 'src/__tests__/utils-for-testing/aria-labels/aria-label-button';
import arialabelCheckbox from 'src/__tests__/utils-for-testing/aria-labels/aria-label-checkbox';
import initialStates from 'src/__tests__/utils-for-testing/data/initial-states';
import reactTestingLibrary from 'src/__tests__/utils-for-testing/react-testing-library-render/react-testing-library-render-functions';
import LocalStorageUtils from 'src/utils/localStorage.utils';
describe('keychainify.component tests:\n', () => {
  afterEach(() => {
    jest.clearAllMocks();
    jest.resetModules();
    cleanup();
  });
  beforeEach(async () => {
    await reactTestingLibrary.renderWithConfiguration(
      <App />,
      initialStates.iniStateAs.defaultExistent,
      {
        app: {
          ledgerRelated: {
            LedgerUtils: {
              isLedgerSupported: true,
            },
          },
        },
      },
    );
    await act(async () => {
      await userEvent.click(screen.getByLabelText(ariaLabelButton.menu));
      await userEvent.click(
        screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.SETTINGS),
      );
      await userEvent.click(
        screen.getByLabelText(ariaLabelButton.menuPreFix + Icons.LINK),
      );
    });
  });
  it('Must load component and match show intro message', () => {
    expect(
      screen.getByLabelText(`${Screen.SETTINGS_KEYCHAINIFY}-page`),
    ).toBeInTheDocument();
    expect(
      screen.getByText(chrome.i18n.getMessage('popup_html_keychainify_text'), {
        exact: true,
      }),
    ).toBeInTheDocument();
  });

  it('Must call saveValueInLocalStorage', async () => {
    const sSaveValueInLocalStorage = jest.spyOn(
      LocalStorageUtils,
      'saveValueInLocalStorage',
    );
    await act(async () => {
      await userEvent.click(
        screen.getByLabelText(arialabelCheckbox.keychainify.checkbox),
      );
    });
    expect(sSaveValueInLocalStorage.mock.lastCall[0]).toBe(
      LocalStorageKeyEnum.KEYCHAINIFY_ENABLED,
    );
    sSaveValueInLocalStorage.mockRestore();
  });
});
