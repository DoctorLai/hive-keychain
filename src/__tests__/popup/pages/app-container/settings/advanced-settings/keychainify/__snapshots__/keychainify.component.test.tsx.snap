// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keychainify.component tests:
 Must load component and match snapshot: keychainify.component Default 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Keychainify
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          aria-label="keychainify-page"
          class="keychainify-page"
        >
          <div
            class="intro"
          >
            This feature allows you to trigger a Hive Keychain operation when clicking on a supported link on a web page or accessing the URL directly in a browser tab. Currently supports the following Hivesigner links: transfers, delegation, custom JSON, witness proxy and votes.
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-keychainify-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-keychainify-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-keychainify"
                class="label"
              >
                Enable Keychainify
              </div>
            </div>
            <div
              class="hint"
            >
              When enabled, this feature will open a Keychain pop when you click on supported links.
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
